### ISSUE TITLE
<pre> tag rendiring issue

### ISSUE THREAD

**Issue by JGiard** [Kozea/WeasyPrint#813]
Hello,
I have a issue when rendering the &lt;pre&gt; under certain conditions.
It looks like that having &lt;br&gt; tags inside them breaks the rendering and the content is cut half-way through. I suppose it might also be a max-line-length issue in the renderer.

I have attached an example showing this, the text is identical but the first is formatted with  &lt;br&gt; tags and the other with regular line breaks.

[test-pre.zip](https://github.com/Kozea/WeasyPrint/files/2905986/test-pre.zip)

I am using weasyprint 45 with cairo 1.15.10

**Comment by Tontyna** [comment]
Ah! That's a nice one!
Not reasoning about a one-line-white-space-pre element with `<br>` instead of newlines ... fact is: the *bad*  text is rendered, but all the invisible TextBoxes have the same `position_y` like the last visible one, their `position_x` increases and increases and increases....
Looks like *\<br>-outside-the-margin* gets lost in (white?) space.

![nr813](https://user-images.githubusercontent.com/35614937/53438565-704cc080-3a00-11e9-91aa-e57fde5817ac.png)

Snippet to reproduce:

```html
<html>
<head>
<style>
p{
  white-space:pre;
  font-size:2em;
}
@page{
  size: A5;
  border: 1px solid black;
  margin-right:5cm; 
}
</style>    
</head>
<body>
<p>
beyond the margin <br>there is no <br> anymore
</p>
</body>
</html>
```

**Comment by liZe** [comment]
OK. The bug happens when preformatted text overfows its parent and when a tag's text ends with a `\n` character.

The bug has been introduced in #557's fix (995a9a3). In `split_first_line`, we always set the `resume_at` value to `None` when the whole text fits in the first line, but that's an error: when the text ends with `\n` for example, we want to remember the forced end of line and thus keep `resume_at`.

(By the way, original example in #557 now fails because of flex :cry:.)

The bug was hidden until the optimization in 22d14ee3 (done after playing with #483) had been added. As there's no automatic line break with preformatted text, there's no need to check that lines can be broken as forced end of lines are supposed to be handled by `split_first_line`.

I've added extra tests to avoid another regression.
